### PR TITLE
Manage content statuses

### PR DESCRIPTION
**Pull request by kont-noor**
### Summary
* Add `content_status` text attribute
* Add `aasm` for content status
* Add `publish`, `draft`, `block`, `unblock` functionality
* Cover with tests
* Some tests cleanup

### Rspec
```
Finished in 3.54 seconds (files took 1.86 seconds to load)
63 examples, 0 failures
Coverage report generated for RSpec to /Users/kont/projects/drevo/theblog/coverage. 211 / 227 LOC (92.95%) covered.
```

### CODE REVIEW

File: app/controllers/theblog/home_controller.rb [hunk 1]
```diff
@@ -3,10 +3,10 @@ class HomeController < Theblog::ApplicationController
     skip_before_action :authenticate_user!
 
     def index
-      @node = Page.find_by(slug: 'home')
-      @node ||= Page.first
+      @node = Page.published.find_by(slug: 'home')
+      @node ||= Page.published.first
 
-      @posts = Post.all
+      @posts = Post.published.all
```
Comment (DevMike) [inline app/controllers/theblog/home_controller.rb]: does 'all' required here?

File: spec/support/shared_examples/features/admin_content_nodes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
+require 'rails_helper'
+
+RSpec.shared_examples :admin_content_node do
+  let(:node_button) { node_name.to_s.humanize }
+  let!(:subject_node) { FactoryGirl.create node_name, author: account }
```
Comment (DevMike) [inline spec/support/shared_examples/features/admin_content_nodes_spec.rb]: let's separate 'let' and 'let!' with empty lines as different logic blocks for better formatting

File: spec/support/shared_examples/features/admin_content_nodes_spec.rb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  let!(:blocked_node) { FactoryGirl.create "blocked_#{node_name}", author: account }
+
+  before do
+    another_account = FactoryGirl.create :confirmed_account
+    FactoryGirl.create node_name, author: another_account
+
+    login_with(account)
```
Comment (DevMike) [inline spec/support/shared_examples/features/admin_content_nodes_spec.rb]: doesn't see any sense to repeat this before every test. It' necessary only once, for others we can authorize user manually to speed up specs

Comment (kont-noor) [inline spec/support/shared_examples/features/admin_content_nodes_spec.rb]: Could you explain this?

Comment (DevMike) [inline spec/support/shared_examples/features/admin_content_nodes_spec.rb]: sure. Your method 'login_with'  does opening login form, entering data, etc. It's no use doing it in every test of this spec, one time is sufficient. In other ones you may use devise 'sign_in' method

**Pull request merged**